### DISCUSSION TITLE
simple short words instead of long character combinations?

### DISCUSSION THREAD

**Discussion by jaju0** [qutebrowser/qutebrowser#7280] [Ideas] [answered]
hey, I'm using qutebrowser for the last few days and it's great so far. It fits all my needs and even more on a tiling window manager, thanks for that I guess, haha. Neverless I found it a bit unintuitive on pages with lots of links to type longer character combinations for the selection. Maybe it's just me and my poor muscle memory but I would find it more convenient to use simple words or syllables for more than 2 chars like 'you', 'me', 'my', 'near', 'far' etc. It's typed more often and is therefore more intuitive, I think. Also, it would be better for typos to not accidentally select the wrong link, like 'jlk' instead of 'jkl'.

Just wanted to share this idea... great project keep it up!

**Comment by The-Compiler** [comment, marked as answer]
Set `hints.mode` to `words` :wink:

**Reply by jaju0** [reply]
thanks

**Reply by Diti** [reply]
Which package is the recommended one for the wordlist? The main one seemingly used in most Linux distributions seems to be `scowl`, but using `${pkgs.scowl}/share/dict/words.txt` in NixOS results in an error from qutebroswer, because the wordfile is not in UTF-8.

**Reply by The-Compiler** [reply]
Debian [distributes](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=775480) the scowl wordlist as UTF-8 and [so does Archlinux](https://gitlab.archlinux.org/archlinux/packaging/packages/words/-/blob/main/PKGBUILD?ref_type=heads#L49). I'd say UTF-8 should be a safe assumption to make nowadays and NixOS should do the same for their package. 

Also see https://github.com/en-wl/wordlist/issues/69.

I suppose you could always run iconv over it and use the resulting file.

**Reply by Diti** [reply]
For anyone interested, `scowl` historically builds its wordlists in ISO-8859 because that’s what `aspell` – one of its build targets – requires.

So there’s no getting around it – in NixOS, the `scowl` package referred to in the configuration needs to be a custom derivation, for example like in the configuration below.

```nix
programs.qutebrowser = let
  scowlUtf8 = pkgs.scowl.overrideAttrs (oldAttrs: {
    postInstall =
      (oldAttrs.postInstall or "")
      + ''
        mkdir --parents "$out/share/dict"
        iconv --from-code=ISO-8859-1 --to-code=UTF-8 "$out/share/dict/words.txt" > "$out/share/dict/words.utf8.txt"
  '';
    nativeBuildInputs = (oldAttrs.nativeBuildInputs or []) ++ [pkgs.glibcLocales];
  });
in {
  enable = true;
  extraConfig = ''
    # . . .
    c.hints.dictionary = '${scowlUtf8}/share/dict/words.utf8.txt'
    c.hints.mode = 'word'
    # . . .
  '';
};
```
